pygsti/modelmembers/torchable.py

```diff
@@ -0,0 +1,50 @@
+"""
+Defines the interface that ModelMembers must satisfy to be compatible with
+the PyTorch-backed forward simulator in pyGSTi/forwardsims/torchfwdsim.py.
+"""
+#***************************************************************************************************
+# Copyright 2024, National Technology & Engineering Solutions of Sandia, LLC (NTESS).
+# Under the terms of Contract DE-NA0003525 with NTESS, the U.S. Government retains certain rights
+# in this software.
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.  You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0 or in the LICENSE file in the root pyGSTi directory.
+#***************************************************************************************************
+
+
+from __future__ import annotations
+from typing import Tuple, TYPE_CHECKING
+if TYPE_CHECKING:
+    import torch as _torch
+
+from pygsti.modelmembers.modelmember import ModelMember
+
+
+class Torchable(ModelMember):
+
+    def stateless_data(self) -> Tuple:
+        """
+        Return this ModelMember's data that is considered constant for purposes of model fitting.
+
+        Note: the word "stateless" here is used in the sense of object-oriented programming.
+        """
+        raise NotImplementedError()   
```

Comment:
No action items, but what is your stance on this strategy of raising NotImplemented over ABC/@abstractmethod and pass?

Reply:
I don't have much experience with ABC/abstractmethod myself. I just looked into it and it seems the main difference is when an error would be raised.

* If you use ABC/abstractmethod then an error will be raised as soon as Python detects that a subclass fails to provide its own implementation of the method.
* If you use the NotImplementedError approach then an error would be raised if (and only if) the function was called at runtime.

Reply:
Yea that's my understanding too, I've been playing around with ABC more recently so was just farming for more opinions :)